x/lazyledgerapp/client/builder.go

```diff
@@ -0,0 +1,182 @@
+package client
+
+import (
+	"context"
+
+	sdkclient "github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	sdktypes "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/tx"
+	"github.com/cosmos/cosmos-sdk/types/tx/signing"
+	authsigning "github.com/cosmos/cosmos-sdk/x/auth/signing"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	"github.com/lazyledger/lazyledger-app/app/params"
+	"google.golang.org/grpc"
+)
+
+// KeyringSigner uses a keyring to sign and build lazyledger-app transactions
+type KeyringSigner struct {
+	keyring.Keyring
+	keyringAccName string
+	accountNumber  uint64
+	sequence       uint64
+	chainID        string
+	encCfg         params.EncodingConfig
+}
```

Comment:
Apologies for the late refactor... I think this slight change makes the function of the code a tad clearer.

Reply:
I think the new name is good. API is very flexible so it allows both straight-forward and advanced scenarios.